### PR TITLE
feat: initial token selector for pay with any token

### PR DESCRIPTION
**Pull request by JackShort**
* this is the initial token selector for pay with any token
* it should currently show the balance for all tokens you own with a balance greater than 0 in the default token list
* you currently cannot select

Screen Shot 2023-01-18 at 2 42 25 PM

### CODE REVIEW

File: src/nft/components/bag/BagTokenSelectorModal.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import { Trans } from '@lingui/macro'
```
Comment (cbachmeier) [inline src/nft/components/bag/BagTokenSelectorModal.tsx:1]: the bag folder is getting fairly large, can we add a folder for the token selector components?

File: src/nft/components/bag/BagTokenSelectorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            return <CurrencyRow key={currency.isToken ? currency.wrapped.address : currency.name} currency={currency} />
+          })}
+        </TokenSelectorContainer>
+      </ModalWrapper>
+      <Overlay onClick={overlayClick} />
+    </Portal>
+  )
+}
+
+const CurrencyRow = ({ currency }: { currency: Currency }) => {
```
Comment (cbachmeier) [inline src/nft/components/bag/BagTokenSelectorModal.tsx:112]: Following uni practices, can we make this its own file?

**Review by cbachmeier — APPROVED** [review]
Works well, minor nits

File: src/nft/components/bag/BagTokenSelectorModal.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import styled from 'styled-components/macro'
+import { ThemedText } from 'theme'
+import { Z_INDEX } from 'theme/zIndex'
+
+const ModalWrapper = styled(Column)`
+  position: fixed;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  width: 420px;
```
[the remark this answers is not in the document]
Comment (JackShort) [reply src/nft/components/bag/BagTokenSelectorModal.tsx:24]: there are no mobile designs for this yet so I am just going to wait on it until I get them

File: src/nft/components/bag/BagTokenSelectorModal.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  const defaultTokens = useAllTokens()
+  const [balances, balancesAreLoading] = useAllTokenBalances()
+  const sortedTokens: Token[] = useMemo(
+    () =>
+      !balancesAreLoading
+        ? Object.values(defaultTokens)
+            .filter((token) => {
+              return balances[token.address]?.greaterThan(0)
+            })
+            .sort(tokenComparator.bind(null, balances))
```
[the remark this answers is not in the document]
Comment (JackShort) [reply src/nft/components/bag/BagTokenSelectorModal.tsx:78]: there is currently a convo about that we are trying to figure it out

File: src/nft/components/bag/BagTokenSelectorModal.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+export const BagTokenSelectorModal = ({ overlayClick }: { overlayClick: () => void }) => {
+  const defaultTokens = useAllTokens()
+  const [balances, balancesAreLoading] = useAllTokenBalances()
+  const sortedTokens: Token[] = useMemo(
+    () =>
+      !balancesAreLoading
+        ? Object.values(defaultTokens)
+            .filter((token) => {
+              return balances[token.address]?.greaterThan(0)
```
[the remark this answers is not in the document]
Comment (JackShort) [reply src/nft/components/bag/BagTokenSelectorModal.tsx:76]: its actually not being checked here. its being checked below from the wrapped native - I will try to come up with a solution for it

**Pull request merged**